app/screens/terms_of_service/terms_of_service.js

```diff
@@ -0,0 +1,247 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {PureComponent} from 'react';
+import PropTypes from 'prop-types';
+import {
+    InteractionManager,
+    ScrollView,
+} from 'react-native';
+import {intlShape} from 'react-intl';
+
+import {getTermsOfService, updateTermsOfServiceStatus} from 'app/actions/views/terms_of_service';
+
+import Loading from 'app/components/loading';
+import Markdown from 'app/components/markdown';
+import ErrorBanner from 'app/components/error_banner';
+import StatusBar from 'app/components/status_bar';
+import {getMarkdownTextStyles, getMarkdownBlockStyles} from 'app/utils/markdown';
+import {changeOpacity, makeStyleSheetFromTheme, setNavigatorStyles} from 'app/utils/theme';
+
+export default class TermsOfService extends PureComponent {
+    static propTypes = {
+        actions: PropTypes.shape({
+            logout: PropTypes.func.isRequired,
+        }),
+        closeButton: PropTypes.object,
+        navigator: PropTypes.object,
+        theme: PropTypes.object,
+    };
+
+    static contextTypes = {
+        intl: intlShape,
+    };
+
+    static defaultProps = {
+        termsEnabled: true,
+    };
+
+    leftButton = {
+        id: 'close-terms-of-service',
+    };
+
+    rightButton = {
+        id: 'accept-terms-of-service',
+        showAsAction: 'always',
+    };
+
+    constructor(props, context) {
+        super(props);
+
+        this.state = {
+            error: null,
+            loading: true,
+            serverError: null,
+            termsId: '',
+            termsText: '',
+        };
+
+        this.rightButton.title = context.intl.formatMessage({id: 'terms_of_service.agreeButton', defaultMessage: 'I Agree'});
+        this.leftButton.icon = props.closeButton;
+
+        props.navigator.setOnNavigatorEvent(this.onNavigatorEvent);
+        this.setNavigatorButtons();
+    }
+
+    componentDidMount() {
+        this.getTerms();
+    }
+
+    componentDidUpdate(prevProps) {
+        if (this.props.theme !== prevProps.theme) {
+            setNavigatorStyles(this.props.navigator, this.props.theme);
+        }
+    }
+
+    setNavigatorButtons = (enabled = true) => {
+        const buttons = {
+            leftButtons: [{...this.leftButton, disabled: !enabled}],
+            rightButtons: [{...this.rightButton, disabled: !enabled}],
+        };
+
+        this.props.navigator.setButtons(buttons);
+    };
+
+    getTerms = () => {
+        const {intl} = this.context;
+
+        this.setState({
+            termsId: '',
+            termsText: '',
+            loading: true,
+            serverError: null,
+        });
+
+        getTermsOfService(
+            (data) => {
+                this.setState({
+                    termsId: data.id,
+                    termsText: data.text,
+                    loading: false,
+                });
+            },
+            () => {
+                this.setState({
```

Comment:
So if for some reason we failed to get the terms from the server there is no way to retry to get them?

then as the code it is now, I could close the modal that logs me out, or Agree to the terms that failed to load, that to me sounds like a bug ;)